torchtitan/datasets/hf_datasets.py

```diff
@@ -72,47 +61,56 @@ def __init__(
         rank: int = 0,
         infinite: bool = False,
     ) -> None:
-        # allow user to pass in a (local or HF hub) path to use unsupported datasets
+        # Force lowercase for consistent comparison
+        dataset_name = dataset_name.lower()
+
         if dataset_name not in _supported_datasets:
```

Comment:
the tutorial didn't mention that we need to update the `_supported_datasets` with new dataset and its path, so it would fail here. I'd recommend we always let user specify the triple, including a default path (but still with the `dataset_path` overriding option).

Reply:
I took your feedback and it was helpful to refactor the code and make it much simpler